gsauthof/dracut-sshd
# dracut-sshd-networkmanager subpackage

This module enables dracut network-manager settings for dracut-sshd.

- If no configuration is provided it uses DHCP to bring up ethernet
  in the same manner as rootfs NetworkManager.
- Cleanly tears down networking prior to switchroot to avoid conflicts
  thereby allowing the OS full control of networking config.
- Network settings could be overriden by copying ifcfg or nmconnection
  settings into the initrd. e.g. static IP's

Fixes: Issues #63 #68 (the latter is a very serious bug with systemd-networkd)
This PR is built on top of #67. I expect #67 is uncontroversial while you maybe want this PR to be adjusted.

### Why the new subpackage?
An rpm could easily auto-configure networkmanager to just work as expected for dracut-sshd. However I anticipate that type of change might be too much of a surprise to some users. Hence the separate subpackage applies the necessary config and makes it work automatically in the majority of situations.

### Tested Operating Systems
Fedora 38
dracut-059-3.fc38.x86_64
NetworkManager-1.42.6-1.fc38.x86_64
openssh-server-9.0p1-15.fc38.x86_64
Rocky 8.8 minimal
Rockky 9.2 minimal

### How to Test
https://fedorapeople.org/~wtogami/a/2023/dracut-sshd-successful-networkmanager-shutdown-before-switchroot.txt
Check `journalctl -b` after a successful boot. You should see NetworkManager and sshd deactivating prior to switchroot like in this log.
